src/domain/opendatafabric/src/dtos/dtos_extra.rs

```diff
@@ -169,6 +174,25 @@ impl TransformSql {
     }
 }
 
+////////////////////////////////////////////////////////////////////////////////
+// ReadStep
+////////////////////////////////////////////////////////////////////////////////
+
+impl ReadStep {
+    pub fn schema(&self) -> Option<&Vec<String>> {
+        match self {
+            ReadStep::Csv(v) => v.schema.as_ref(),
+            ReadStep::Json(v) => v.schema.as_ref(),
+            ReadStep::JsonLines(v) => v.schema.as_ref(),
+            ReadStep::NdJson(v) => v.schema.as_ref(),
+            ReadStep::GeoJson(v) => v.schema.as_ref(),
+            ReadStep::NdGeoJson(v) => v.schema.as_ref(),
+            ReadStep::EsriShapefile(v) => v.schema.as_ref(),
+            ReadStep::Parquet(v) => v.schema.as_ref(),
```

Comment:
Note, this code is shouting `schema` field is stored at a wrong hierarchy level

Reply:
This is a "poor man's base class" :)

Looking at metadata structure I think `schema` property is well positioned:
```yaml
read:
  kind: csv
  schema: ...
```

I wouldn't want adding another level of nesting there just for field reuse.